zebra-consensus/src/block/subsidy/funding_streams.rs

```diff
@@ -50,3 +48,116 @@ pub fn funding_stream_values(
+    //   This is the same as `floor()`, because these numbers are all positive.
+    (height.0 + (POST_BLOSSOM_HALVING_INTERVAL.0) - (height_for_halving(network).0))
+        / (FUNDING_STREAM_ADDRESS_CHANGE_INTERVAL.0)
+}
+
+/// Returns the position in the address slice for each funding stream
+/// as described in [protocol specification §7.10][7.10]
+///
+/// [7.10]: https://zips.z.cash/protocol/protocol.pdf#fundingstreams
+fn funding_stream_address_index(height: Height, network: Network) -> usize {
+    let num_addresses = match network {
+        Network::Mainnet => FUNDING_STREAMS_NUM_ADDRESSES_MAINNET,
+        Network::Testnet => FUNDING_STREAMS_NUM_ADDRESSES_TESTNET,
+    };
+
+    let index = 1u32
+        .checked_add(funding_stream_address_period(height, network))
+        .expect("no overflow should happen in this sum")
+        .checked_sub(funding_stream_address_period(
+            FUNDING_STREAM_HEIGHT_RANGES.get(&network).unwrap().start,
+            network,
+        ))
+        .expect("no overflow should happen in this sub") as usize;
+
+    assert!(index > 0 && index <= num_addresses);
+    // spec formula will output an index starting at 1 but
+    // Zebra indices for addresses start at zero, return converted.
+    index - 1
+}
+
+/// Return the address corresponding to given height, network and funding stream receiver.
+pub fn funding_stream_address(
+    height: Height,
+    network: Network,
+    receiver: FundingStreamReceiver,
+) -> Address {
+    let index = funding_stream_address_index(height, network);
+    let address = &FUNDING_STREAM_ADDRESSES
+        .get(&network)
+        .expect("there is always another hash map as value for a given valid network")
+        .get(&receiver)
+        .expect("in the inner hash map there is always a vector of strings with addresses")[index];
+    Address::from_str(address).expect("Address should deserialize")
+}
+
+/// Given a founders reward address, create a script and check if it is the same
+/// as the given lock_script as described in [protocol specification §7.10][7.10]
+///
+/// [7.10]: https://zips.z.cash/protocol/protocol.pdf#fundingstreams.
+pub fn check_script_form(lock_script: Script, address: Address) -> bool {
+    let mut address_hash = address
+        .zcash_serialize_to_vec()
+        .expect("we should get address bytes here");
+
+    address_hash = address_hash[2..22].to_vec();
+    address_hash.insert(0, OpCode::Push20Bytes as u8);
+    address_hash.insert(0, OpCode::Hash160 as u8);
+    address_hash.insert(address_hash.len(), OpCode::Equal as u8);
+    if lock_script.as_raw_bytes().len() == address_hash.len()
+        && lock_script == Script::new(&address_hash)
+    {
+        return true;
+    }
+    false
+}
+
+/// Returns a list of outputs in `Transaction`, which have a script address equal to `Address`
+/// and a value equal to `Amount`.
+pub fn find_output_with_address_and_amount(
```

Comment:
Naming: In Rust, `find` returns a single value or `None`. But `filter` returns all matching values.
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter

```suggestion
pub fn filter_outputs_by_address_and_amount(
```

Reply:
Thanks, `filter_outputs_by_address_and_amount` was removed but as part of https://github.com/ZcashFoundation/zebra/commit/0f43eaf2b4ad4b2090db40b34f933decd96fbf87 we now have `filter_outputs_by_address`